### ISSUE TITLE
Syntax highlighting

### ISSUE THREAD

**Issue by akubiczek** [irmen/prog8#23]
Hello!

I have developed an extension for Visual Studio Code to proper Prog8 syntax highlighting. It's far from being perfect but it helps a lot: https://github.com/akubiczek/Prog8-TmLanguage-VsCode

![Sample](https://i.ibb.co/zPddNxx/Zrzut-ekranu-2020-12-2-o-14-08-09.png)

There is also a regular Textmate XML that can be used in other tools.

best regards
Adam

**Comment by irmen** [comment]
That is great! I will mention this in the documentation

I have been thinking about creating syntax highlighters for KDE Kate, Vim and Sublime Text but it is a lot of work....

**Comment by akubiczek** [comment]
Very well :) I just made some improvements and added a yaml file for Sublime Text 3: https://github.com/akubiczek/Prog8-TmLanguage-VsCode/blob/master/src/prog8.sublime3.yaml

It's not tested though - I don't use Sublime.

**Comment by irmen** [comment]
Hm, can't get the sublime syntax to work yet... appreciate the effort though. Could be something on my end. I'm not familiar yet with syntax def files for it too.

**Comment by akubiczek** [comment]
Hello, the new release comes with Sublime 3 extension. Happy coding ;)

**Comment by Elektron72** [comment]
I'm currently working on a Prog8 syntax file for Vim. It's not done yet, but here's a screenshot:
![Screenshot from 2021-03-22 10-32-24](https://user-images.githubusercontent.com/36974021/112006742-6112a100-8afa-11eb-9197-7163035a746a.png)
When it is ready for use, should I fork this repo, place it in syntax-files/Vim, and submit a pull request, or should I create a new repo?

**Comment by irmen** [comment]
please use a PR!  Thanks for your work on this !

**Comment by meisl** [comment]
Hi,

I just made a syntax file for Notepad++.
Without `asm {{` ... `}}` parts though, unsure how to go about that.
So, it's not finished but I can make a pull-request already if you want.

![screenshot](https://user-images.githubusercontent.com/1493886/117060262-025b5c80-ad21-11eb-8cc2-366cd743febb.png)

**Comment by irmen** [comment]
yeah please make a PR! It's great already even if not covering everything!

**Comment by meisl** [comment]
Done: https://github.com/irmen/prog8/pull/32

**Comment by adiee5** [comment]
Did a pr for `nano` highlighting! #123

**Comment by irmen** [comment]
updated the VSCode link to new plugin maintained by Thraka
